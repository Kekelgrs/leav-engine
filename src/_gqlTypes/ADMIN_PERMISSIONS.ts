/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: ADMIN_PERMISSIONS
// ====================================================

export interface ADMIN_PERMISSIONS_adminPermissions {
    admin_access_attributes: boolean;
    admin_access_libraries: boolean;
    admin_access_permissions: boolean;
    admin_access_trees: boolean;
    admin_create_attribute: boolean;
    admin_create_library: boolean;
    admin_create_tree: boolean;
    admin_delete_attribute: boolean;
    admin_delete_library: boolean;
    admin_delete_tree: boolean;
    admin_edit_attribute: boolean;
    admin_edit_library: boolean;
    admin_edit_permission: boolean;
    admin_edit_tree: boolean;
}

export interface ADMIN_PERMISSIONS {
    adminPermissions: ADMIN_PERMISSIONS_adminPermissions | null;
}
