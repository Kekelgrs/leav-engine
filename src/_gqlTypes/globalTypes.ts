/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

export enum ActionIOTypes {
    boolean = 'boolean',
    number = 'number',
    object = 'object',
    string = 'string'
}

export enum AttributeFormat {
    boolean = 'boolean',
    date = 'date',
    encrypted = 'encrypted',
    extended = 'extended',
    numeric = 'numeric',
    text = 'text'
}

export enum AttributeType {
    advanced = 'advanced',
    advanced_link = 'advanced_link',
    simple = 'simple',
    simple_link = 'simple_link',
    tree = 'tree'
}

export enum AvailableActionsName {
    encrypt = 'encrypt',
    formatDate = 'formatDate',
    formatNumber = 'formatNumber',
    maskValue = 'maskValue',
    parseJSON = 'parseJSON',
    toBoolean = 'toBoolean',
    toJSON = 'toJSON',
    toNumber = 'toNumber',
    toString = 'toString',
    validateFormat = 'validateFormat',
    validateRegex = 'validateRegex'
}

export enum AvailableLanguage {
    en = 'en',
    fr = 'fr'
}

export enum PermissionTypes {
    admin = 'admin',
    attribute = 'attribute',
    library = 'library',
    record = 'record'
}

export enum PermissionsActions {
    access = 'access',
    access_attribute = 'access_attribute',
    admin_access_attributes = 'admin_access_attributes',
    admin_access_libraries = 'admin_access_libraries',
    admin_access_navigator = 'admin_access_navigator',
    admin_access_permissions = 'admin_access_permissions',
    admin_access_trees = 'admin_access_trees',
    admin_create_attribute = 'admin_create_attribute',
    admin_create_library = 'admin_create_library',
    admin_create_tree = 'admin_create_tree',
    admin_delete_attribute = 'admin_delete_attribute',
    admin_delete_library = 'admin_delete_library',
    admin_delete_tree = 'admin_delete_tree',
    admin_edit_attribute = 'admin_edit_attribute',
    admin_edit_library = 'admin_edit_library',
    admin_edit_permission = 'admin_edit_permission',
    admin_edit_tree = 'admin_edit_tree',
    create = 'create',
    create_value = 'create_value',
    delete = 'delete',
    delete_value = 'delete_value',
    edit = 'edit',
    edit_value = 'edit_value'
}

export enum PermissionsRelation {
    and = 'and',
    or = 'or'
}

export enum ValueVersionMode {
    simple = 'simple',
    smart = 'smart'
}

export interface ActionConfigurationInput {
    name: AvailableActionsName;
    params?: (ActionConfigurationParamInput | null)[] | null;
}

export interface ActionConfigurationParamInput {
    name: string;
    value: string;
}

export interface ActionsListConfigurationInput {
    saveValue?: (ActionConfigurationInput | null)[] | null;
    getValue?: (ActionConfigurationInput | null)[] | null;
    deleteValue?: (ActionConfigurationInput | null)[] | null;
}

export interface AttributeInput {
    id: string;
    type?: AttributeType | null;
    format?: AttributeFormat | null;
    label?: SystemTranslationInput | null;
    linked_library?: string | null;
    linked_tree?: string | null;
    embedded_fields?: (EmbeddedAttributeInput | null)[] | null;
    actions_list?: ActionsListConfigurationInput | null;
    permissions_conf?: Treepermissions_confInput | null;
    multiple_values?: boolean | null;
    versions_conf?: ValuesVersionsConfInput | null;
}

export interface EmbeddedAttributeInput {
    id: string;
    format?: AttributeFormat | null;
    label?: SystemTranslationInput | null;
    validation_regex?: string | null;
    embedded_fields?: (EmbeddedAttributeInput | null)[] | null;
}

export interface LibraryInput {
    id: string;
    label?: SystemTranslationInput | null;
    attributes?: string[] | null;
    permissions_conf?: Treepermissions_confInput | null;
    recordIdentityConf?: RecordIdentityConfInput | null;
}

export interface PermissionActionInput {
    name: PermissionsActions;
    allowed?: boolean | null;
}

/**
 * If users group is not specified, permission will be saved at root level.
 * If saving a tree-based permission (record or attribute) and tree target's id is not specified,
 * permission will be saved at root level for any element of the tree.
 */
export interface PermissionInput {
    type: PermissionTypes;
    applyTo?: string | null;
    usersGroup?: string | null;
    actions: PermissionActionInput[];
    permissionTreeTarget?: PermissionsTreeTargetInput | null;
}

/**
 * Element on which we want to retrieve record or attribute permission. Record ID is mandatory,
 * attributeId is only required for attribute permission
 */
export interface PermissionTarget {
    attributeId?: string | null;
    recordId: string;
}

/**
 * If id and library are not specified, permission will apply to tree root
 */
export interface PermissionsTreeTargetInput {
    tree: string;
    library?: string | null;
    id?: string | null;
}

export interface RecordIdentityConfInput {
    label?: string | null;
    color?: string | null;
    preview?: string | null;
}

export interface SystemTranslationInput {
    fr: string;
    en?: string | null;
}

export interface TreeElementInput {
    id: string;
    library: string;
}

export interface TreeInput {
    id: string;
    libraries?: string[] | null;
    label?: SystemTranslationInput | null;
}

export interface Treepermissions_confInput {
    permissionTreeAttributes: string[];
    relation: PermissionsRelation;
}

export interface ValueBatchInput {
    attribute?: string | null;
    id_value?: string | null;
    value?: string | null;
}

export interface ValueVersionInput {
    name: string;
    value: TreeElementInput;
}

export interface ValuesVersionsConfInput {
    versionable: boolean;
    mode?: ValueVersionMode | null;
    trees?: string[] | null;
}

//==============================================================
// END Enums and Input Objects
//==============================================================
