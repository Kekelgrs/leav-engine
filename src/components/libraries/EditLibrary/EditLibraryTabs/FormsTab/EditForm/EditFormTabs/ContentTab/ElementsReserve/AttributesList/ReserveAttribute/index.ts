import ReserveAttribute from './ReserveAttribute';
export default ReserveAttribute;