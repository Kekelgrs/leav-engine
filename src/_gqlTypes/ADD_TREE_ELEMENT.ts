/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import {TreeElementInput} from './globalTypes';

// ====================================================
// GraphQL mutation operation: ADD_TREE_ELEMENT
// ====================================================

export interface ADD_TREE_ELEMENT_treeAddElement {
    id: string | null;
    library: string | null;
}

export interface ADD_TREE_ELEMENT {
    treeAddElement: ADD_TREE_ELEMENT_treeAddElement;
}

export interface ADD_TREE_ELEMENTVariables {
    treeId: string;
    element: TreeElementInput;
    parent?: TreeElementInput | null;
}
