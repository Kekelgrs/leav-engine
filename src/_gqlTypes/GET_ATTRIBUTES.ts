/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import {AvailableLanguage, AttributeType, AttributeFormat, PermissionsRelation, ValueVersionMode} from './globalTypes';

// ====================================================
// GraphQL query operation: GET_ATTRIBUTES
// ====================================================

export interface GET_ATTRIBUTES_attributes_permissions_conf_permissionTreeAttributes {
    id: string;
    linked_tree: string | null;
    label: any | null;
}

export interface GET_ATTRIBUTES_attributes_permissions_conf {
    permissionTreeAttributes: GET_ATTRIBUTES_attributes_permissions_conf_permissionTreeAttributes[];
    relation: PermissionsRelation;
}

export interface GET_ATTRIBUTES_attributes_versions_conf {
    versionable: boolean;
    mode: ValueVersionMode | null;
    trees: string[] | null;
}

export interface GET_ATTRIBUTES_attributes {
    id: string;
    type: AttributeType;
    format: AttributeFormat | null;
    system: boolean;
    label: any | null;
    linked_library: string | null;
    linked_tree: string | null;
    multiple_values: boolean;
    permissions_conf: GET_ATTRIBUTES_attributes_permissions_conf | null;
    versions_conf: GET_ATTRIBUTES_attributes_versions_conf | null;
}

export interface GET_ATTRIBUTES {
    attributes: GET_ATTRIBUTES_attributes[] | null;
}

export interface GET_ATTRIBUTESVariables {
    lang?: AvailableLanguage[] | null;
    id?: string | null;
    label?: string | null;
    type?: (AttributeType | null)[] | null;
    format?: (AttributeFormat | null)[] | null;
    system?: boolean | null;
    multiple_values?: boolean | null;
    versionable?: boolean | null;
}
