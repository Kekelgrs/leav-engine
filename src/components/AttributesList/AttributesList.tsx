import * as React from 'react';
import {translate, TranslationFunction} from 'react-i18next';
import {Table} from 'semantic-ui-react';
import {GET_ATTRIBUTES_attributes} from '../../_gqlTypes/GET_ATTRIBUTES';

interface IAttributesListProps {
    attributes: GET_ATTRIBUTES_attributes[] | null;
    t: TranslationFunction;
    onRowClick: (attribute: GET_ATTRIBUTES_attributes) => void;
}

function AttributesList({attributes, t, onRowClick}: IAttributesListProps): JSX.Element {
    return attributes !== null ? (
        <Table selectable striped>
            <Table.Header>
                <Table.Row>
                    <Table.HeaderCell>{t('attributes.label')}</Table.HeaderCell>
                    <Table.HeaderCell>{t('attributes.ID')}</Table.HeaderCell>
                    <Table.HeaderCell>{t('attributes.type')}</Table.HeaderCell>
                    <Table.HeaderCell>{t('attributes.format')}</Table.HeaderCell>
                    <Table.HeaderCell />
                </Table.Row>
            </Table.Header>
            <Table.Body>
                {attributes.map(a => {
                    const attrLabel = a.label !== null ? a.label.fr || a.label.en || a.id : a.id;
                    const onClick = () => onRowClick(a);
                    return (
                        <Table.Row key={a.id} onClick={onClick}>
                            <Table.Cell>{attrLabel}</Table.Cell>
                            <Table.Cell>{a.id}</Table.Cell>
                            <Table.Cell>{t('attributes.types.' + a.type)}</Table.Cell>
                            <Table.Cell>{a.format ? t('attributes.formats.' + a.format) : ''}</Table.Cell>
                            <Table.Cell />
                        </Table.Row>
                    );
                })}
            </Table.Body>
        </Table>
    ) : (
        <p>No attributes</p>
    );
}

export default translate()(AttributesList);
