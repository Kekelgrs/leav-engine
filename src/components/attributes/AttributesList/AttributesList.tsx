import * as React from 'react';
import {withNamespaces, WithNamespaces} from 'react-i18next';
import {Checkbox, Dropdown, Input, Table} from 'semantic-ui-react';
import Loading from 'src/components/shared/Loading';
import {localizedLabel} from 'src/utils/utils';
import {AttributeDetails} from 'src/_gqlTypes/AttributeDetails';
import {GET_ATTRIBUTES_attributes} from 'src/_gqlTypes/GET_ATTRIBUTES';
import {AttributeFormat, AttributeType} from 'src/_gqlTypes/globalTypes';

interface IAttributesListProps extends WithNamespaces {
    attributes: AttributeDetails[] | null;
    onRowClick: (attribute: GET_ATTRIBUTES_attributes) => void;
    onFiltersUpdate?: (filters: any) => void;
    loading: boolean;
    withFilters?: boolean;
    filters?: any;
}

class AttributesList extends React.Component<IAttributesListProps> {
    public static defaultProps = {
        attributes: [],
        loading: false,
        withFilters: true,
        filters: {}
    };

    constructor(props: IAttributesListProps) {
        super(props);
    }

    public render = () => {
        const {attributes, t, onRowClick, children, loading, filters, withFilters, i18n: i18next} = this.props;
        const childrenList: React.ReactNode[] = children ? (!Array.isArray(children) ? [children] : children) : [];
        const types = Object.keys(AttributeType).map(type => ({
            key: type,
            value: type,
            text: t('attributes.types.' + type)
        }));

        const formats = Object.keys(AttributeFormat).map(format => ({
            key: format,
            value: format,
            text: t('attributes.formats.' + format)
        }));

        return (
            <Table selectable striped>
                <Table.Header>
                    <Table.Row>
                        <Table.HeaderCell width={4}>{t('attributes.label')}</Table.HeaderCell>
                        <Table.HeaderCell width={4}>{t('attributes.ID')}</Table.HeaderCell>
                        <Table.HeaderCell width={3}>{t('attributes.type')}</Table.HeaderCell>
                        <Table.HeaderCell width={3}>{t('attributes.format')}</Table.HeaderCell>
                        <Table.HeaderCell width={1}>{t('attributes.isSystem')}</Table.HeaderCell>
                        <Table.HeaderCell width={1} />
                    </Table.Row>
                    {withFilters && (
                        <Table.Row className="filters">
                            <Table.HeaderCell>
                                <Input
                                    size="small"
                                    fluid
                                    placeholder={t('attributes.label') + '...'}
                                    name="label"
                                    value={filters.label || ''}
                                    onChange={this._handleFilterChange}
                                />
                            </Table.HeaderCell>
                            <Table.HeaderCell>
                                <Input
                                    size="small"
                                    fluid
                                    placeholder={t('attributes.ID') + '...'}
                                    name="id"
                                    value={filters.id || ''}
                                    onChange={this._handleFilterChange}
                                />
                            </Table.HeaderCell>
                            <Table.HeaderCell>
                                <Dropdown
                                    fluid
                                    multiple
                                    search
                                    selection
                                    options={types}
                                    placeholder={t('attributes.type') + '...'}
                                    name="type"
                                    value={filters.type || []}
                                    onChange={this._handleFilterChange}
                                />
                            </Table.HeaderCell>
                            <Table.HeaderCell>
                                <Dropdown
                                    fluid
                                    multiple
                                    search
                                    selection
                                    options={formats}
                                    placeholder={t('attributes.format') + '...'}
                                    name="format"
                                    value={filters.format || []}
                                    onChange={this._handleFilterChange}
                                />
                            </Table.HeaderCell>
                            <Table.HeaderCell>
                                <Checkbox
                                    indeterminate={typeof filters.system === 'undefined'}
                                    name="system"
                                    checked={filters.system}
                                    onChange={this._handleFilterChange}
                                />
                            </Table.HeaderCell>
                            <Table.HeaderCell />
                        </Table.Row>
                    )}
                </Table.Header>
                <Table.Body>
                    {loading ? (
                        <Table.Row>
                            <Table.Cell colSpan={6}>
                                <Loading />
                            </Table.Cell>
                        </Table.Row>
                    ) : (
                        !!attributes &&
                        attributes.map(a => {
                            const attrLabel = localizedLabel(a.label, i18next);
                            const onClick = () => onRowClick(a);
                            return (
                                <Table.Row key={a.id} onClick={onClick}>
                                    <Table.Cell>{attrLabel}</Table.Cell>
                                    <Table.Cell>{a.id}</Table.Cell>
                                    <Table.Cell>{t('attributes.types.' + a.type)}</Table.Cell>
                                    <Table.Cell>{a.format ? t('attributes.formats.' + a.format) : ''}</Table.Cell>
                                    <Table.Cell width={1}>
                                        <Checkbox readOnly checked={a.system} />
                                    </Table.Cell>
                                    <Table.Cell textAlign="right" width={1} className="actions">
                                        {childrenList.map(child =>
                                            React.cloneElement(child as React.ReactElement<any>, {attribute: a})
                                        )}
                                    </Table.Cell>
                                </Table.Row>
                            );
                        })
                    )}
                </Table.Body>
            </Table>
        );
    }

    private _handleFilterChange = (e: React.SyntheticEvent, d: any) => {
        // If a checkbox was not checked and is clicked, go back to indeterminate state
        if (d.type === 'checkbox' && this.props.filters.system === false && d.checked) {
            d.indeterminate = true;
            d.checked = false;
        }

        if (this.props.onFiltersUpdate) {
            this.props.onFiltersUpdate(d);
        }
    }
}

export default withNamespaces()(AttributesList);
