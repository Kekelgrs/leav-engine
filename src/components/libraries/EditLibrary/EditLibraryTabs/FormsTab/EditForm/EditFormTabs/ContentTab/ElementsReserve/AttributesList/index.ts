import AttributesList from './AttributesList';
export default AttributesList;