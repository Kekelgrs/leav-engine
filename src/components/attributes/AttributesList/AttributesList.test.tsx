import {mount, shallow} from 'enzyme';
import React from 'react';
import {MockedProvider} from 'react-apollo/test-utils';
import {GET_ATTRIBUTES_attributes} from '../../../_gqlTypes/GET_ATTRIBUTES';
import {AttributeFormat, AttributeType} from '../../../_gqlTypes/globalTypes';
import AttributesList from './AttributesList';

describe('AttributesList', () => {
    const attributes: GET_ATTRIBUTES_attributes[] = [
        {
            id: 'attr1',
            type: AttributeType.simple,
            format: AttributeFormat.text,
            system: false,
            label: {fr: 'Test 1', en: null},
            linked_tree: null,
            permissions_conf: null
        },
        {
            id: 'attr2',
            type: AttributeType.simple,
            format: AttributeFormat.text,
            system: false,
            label: {fr: 'Test 2', en: null},
            linked_tree: null,
            permissions_conf: null
        },
        {
            id: 'attr3',
            type: AttributeType.simple,
            format: AttributeFormat.text,
            system: false,
            label: {fr: 'Test 3', en: null},
            linked_tree: null,
            permissions_conf: null
        }
    ];

    const onRowClick = jest.fn();
    const onFiltersUpdate = jest.fn();
    test('Render attributes list with filters', async () => {
        const comp = shallow(
            <MockedProvider>
                <AttributesList
                    loading={false}
                    attributes={attributes}
                    onRowClick={onRowClick}
                    onFiltersUpdate={onFiltersUpdate}
                />
            </MockedProvider>
        );

        const attrListComp = comp.find('AttributesList').shallow();

        expect(attrListComp.find('TableBody TableRow').length).toEqual(3);
        expect(attrListComp.find('TableRow.filters').length).toEqual(1);
    });

    test('Render attributes list without filters', async () => {
        const comp = shallow(
            <MockedProvider>
                <AttributesList
                    loading={false}
                    attributes={attributes}
                    onRowClick={onRowClick}
                    onFiltersUpdate={onFiltersUpdate}
                    withFilters={false}
                />
            </MockedProvider>
        );
        const attrListComp = comp.find('AttributesList').shallow();
        expect(attrListComp.find('TableRow.filters').length).toEqual(0);
    });

    test('Render children in actions cell', async () => {
        const comp = shallow(
            <MockedProvider>
                <AttributesList
                    loading={false}
                    attributes={attributes}
                    onRowClick={onRowClick}
                    onFiltersUpdate={onFiltersUpdate}
                    withFilters={false}
                >
                    <div key="attr_lib_test" className="children_to_render" />
                </AttributesList>
            </MockedProvider>
        );
        const attrListComp = comp.find('AttributesList').shallow();
        expect(attrListComp.find('TableCell.actions .children_to_render').length).toEqual(3);
    });

    test('Render without children in actions cell', async () => {
        const comp = shallow(
            <MockedProvider>
                <AttributesList
                    loading={false}
                    attributes={attributes}
                    onRowClick={onRowClick}
                    onFiltersUpdate={onFiltersUpdate}
                    withFilters={false}
                />
            </MockedProvider>
        );
        const attrListComp = comp.find('AttributesList').shallow();
        expect(attrListComp.find('TableCell.actions').children().length).toEqual(0);
    });

    test('Calls callback on filter update', () => {
        const changeFilter = jest.fn();
        const comp = mount(
            <MockedProvider>
                <AttributesList
                    loading={false}
                    attributes={attributes}
                    onRowClick={onRowClick}
                    onFiltersUpdate={changeFilter}
                />
            </MockedProvider>
        );

        comp.find('.filters input[name="label"]').simulate('change', {target: {value: 'MyLabel'}});
        comp.find('.filters input[name="id"]').simulate('change');
        comp.find('.filters div[name="format"]').simulate('change');
        comp.find('.filters div[name="type"]').simulate('change');
        comp.find('.filters input[name="system"]').simulate('change');

        expect(changeFilter).toHaveBeenCalledTimes(5);
        expect(changeFilter.mock.calls[0][0]).toMatchObject({value: 'MyLabel'});
    });
});
