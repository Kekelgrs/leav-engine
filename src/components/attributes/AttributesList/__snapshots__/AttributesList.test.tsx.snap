// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttributesList Snapshot test 1`] = `
<table
  className="ui selectable striped table"
>
  <thead
    className=""
  >
    <tr
      className=""
    >
      <th
        className="four wide"
      >
        
      </th>
      <th
        className="four wide"
      >
        
      </th>
      <th
        className="three wide"
      >
        
      </th>
      <th
        className="three wide"
      >
        
      </th>
      <th
        className="one wide"
      >
        
      </th>
      <th
        className="one wide"
      />
    </tr>
    <tr
      className="filters"
    >
      <th
        className=""
      >
        <div
          className="ui small fluid input"
        >
          <input
            disabled={undefined}
            name="label"
            onChange={[Function]}
            placeholder="..."
            tabIndex={undefined}
            type="text"
            value=""
          />
        </div>
      </th>
      <th
        className=""
      >
        <div
          className="ui small fluid input"
        >
          <input
            disabled={undefined}
            name="id"
            onChange={[Function]}
            placeholder="..."
            tabIndex={undefined}
            type="text"
            value=""
          />
        </div>
      </th>
      <th
        className=""
      >
        <div
          aria-busy={undefined}
          aria-disabled={undefined}
          aria-expanded={false}
          className="ui fluid multiple search selection dropdown"
          name="type"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="combobox"
          tabIndex={undefined}
        >
          <input
            aria-autocomplete="list"
            autoComplete="off"
            className="search"
            onChange={[Function]}
            style={
              Object {
                "width": undefined,
              }
            }
            tabIndex={0}
            type="text"
            value=""
          />
          <span
            className="sizer"
          />
          <div
            aria-live="polite"
            className="default text"
            role="alert"
          >
            ...
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            aria-multiselectable={true}
            className="menu transition"
            role="listbox"
          >
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={true}
              className="selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
          </div>
        </div>
      </th>
      <th
        className=""
      >
        <div
          aria-busy={undefined}
          aria-disabled={undefined}
          aria-expanded={false}
          className="ui fluid multiple search selection dropdown"
          name="format"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          role="combobox"
          tabIndex={undefined}
        >
          <input
            aria-autocomplete="list"
            autoComplete="off"
            className="search"
            onChange={[Function]}
            style={
              Object {
                "width": undefined,
              }
            }
            tabIndex={0}
            type="text"
            value=""
          />
          <span
            className="sizer"
          />
          <div
            aria-live="polite"
            className="default text"
            role="alert"
          >
            ...
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            aria-multiselectable={true}
            className="menu transition"
            role="listbox"
          >
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={true}
              className="selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
            <div
              aria-checked={false}
              aria-disabled={undefined}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                
              </span>
            </div>
          </div>
        </div>
      </th>
      <th
        className=""
      >
        <div
          className="ui indeterminate fitted checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            disabled={undefined}
            id={undefined}
            name="system"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          />
        </div>
      </th>
      <th
        className=""
      />
    </tr>
  </thead>
  <tbody
    className=""
  >
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        Test 1
      </td>
      <td
        className=""
      >
        attr1
      </td>
      <td
        className=""
      >
        
      </td>
      <td
        className=""
      >
        
      </td>
      <td
        className="one wide"
      >
        <div
          className="ui fitted read-only checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            disabled={undefined}
            id={undefined}
            name={undefined}
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          />
        </div>
      </td>
      <td
        className="right aligned one wide"
      >
        <div
          attribute={
            Object {
              "format": "text",
              "id": "attr1",
              "label": Object {
                "en": null,
                "fr": "Test 1",
              },
              "system": false,
              "type": "simple",
            }
          }
        />
      </td>
    </tr>
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        Test 2
      </td>
      <td
        className=""
      >
        attr2
      </td>
      <td
        className=""
      >
        
      </td>
      <td
        className=""
      >
        
      </td>
      <td
        className="one wide"
      >
        <div
          className="ui fitted read-only checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            disabled={undefined}
            id={undefined}
            name={undefined}
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          />
        </div>
      </td>
      <td
        className="right aligned one wide"
      >
        <div
          attribute={
            Object {
              "format": "text",
              "id": "attr2",
              "label": Object {
                "en": null,
                "fr": "Test 2",
              },
              "system": false,
              "type": "simple",
            }
          }
        />
      </td>
    </tr>
    <tr
      className=""
      onClick={[Function]}
    >
      <td
        className=""
      >
        Test 3
      </td>
      <td
        className=""
      >
        attr3
      </td>
      <td
        className=""
      >
        
      </td>
      <td
        className=""
      >
        
      </td>
      <td
        className="one wide"
      >
        <div
          className="ui fitted read-only checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={false}
            className="hidden"
            disabled={undefined}
            id={undefined}
            name={undefined}
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
            value={undefined}
          />
          <label
            htmlFor={undefined}
          />
        </div>
      </td>
      <td
        className="right aligned one wide"
      >
        <div
          attribute={
            Object {
              "format": "text",
              "id": "attr3",
              "label": Object {
                "en": null,
                "fr": "Test 3",
              },
              "system": false,
              "type": "simple",
            }
          }
        />
      </td>
    </tr>
  </tbody>
</table>
`;
